### DISCUSSION TITLE
P-Q curve reading

### DISCUSSION THREAD

**Discussion by torodecamargue** [nextfoam/baram#82] [General]
Hello,

Could you precise units and correct data writing  of the P-Q curve in csv file ? 

I try this, it seems to be ok for reading by BaramFlow 
![image](https://github.com/nextfoam/baram/assets/75060318/51086a5e-c558-4926-af4a-846e4c3111e5)

but I have the following error. I try steady or transient case it gives the same error.
![image](https://github.com/nextfoam/baram/assets/75060318/7fbd656a-afb0-4921-b125-970e3bd3197e)

The circuit is a closed loop with a single fan. I add some porous jumps BC conditions in the loop. I use a symmetry plane BC.
I don't how to configure this case. Do I need to have 1 single interface for inlet and outlet of the fan or 1 for each ? 

![image](https://github.com/nextfoam/baram/assets/75060318/5998e9f1-5e7b-4152-a52f-2b94d4827f14)

Best regards

Julien

**Comment by torodecamargue** [comment]
Hello,

Maybe you have an example of a closed loop circuit with the fan BC in Baram ?

Best regards

**Comment by kdg5424** [comment]
Hello Julien.

First, you should sort the "pressure" in ascending order in the CSV file.
Second, we don't have the closed loop circuit with the fan BC tutorial.
So if you send us your stl file, we will try with your case.

Thanks.

**Comment by torodecamargue** [comment]
Hello Duggyu,

Thank you. Find attached the STL. int(i) interfaces are the surface for the future monitoring mass flow, pressure etc, add porous jump if I need etc.
[SLT_closed_loop.zip](https://github.com/user-attachments/files/15542675/SLT_closed_loop.zip)


Julien

**Comment by torodecamargue** [comment]
Hello,

Could you post a simple case with the Fan condition used, or a sketch explaining which BC I have to use ? 

For example, in a close loop air circuit, or in an open duct (which BC at inlet, outlet?)

Thank you for your help.

Julien

**Comment by kdg5424** [comment]
Hello Julien. This is Donggyu.

We found a serious problem with FAN BC. I'm really sorry for your inconvenience.
The problem is that if users set FAN BC with internal flow case, it doesn't work.
We'll solve the problem in hurry, after that I'll let you know.
I'm really sorry again.

Best regards

Donggyu kim

**Comment by torodecamargue** [comment]
Hello Donggyu,

Do you find what is wrong with the fan condition? 

Julien

**Comment by kdg5424** [comment]
Hello Julien. In my opinion, it's not so simple problem.
There are several fan boundary conditions in OpenFOAM ESI, like fan, fanPressureJump.
And we're using fan boundary condition in BARAM.
But it has a problem. The fan boundary condition in OpenFOAM, it doesn't use P-Q curve but it uses U-P curve.
So we're discussing it how we should chnage.

Could you wait until then?

Thanks

Regards

Donggyyu kim

**Comment by torodecamargue** [comment]
Hello Donggyu,

I understand.
Yes of course I can wait. Thank you for your work.

BR 

julien

**Comment by torodecamargue** [comment]
Hello Donggyu,

Do you know if the fanBC is updated in the current Baram release  ? 

Best regards

Julien

**Comment by kdg5424** [comment]
Hello Julien. This is Donggyu from NEXTFOAM.

I'm really sorry. But we've not changed it yet.

As I told you the problems are two things.

1) The fan boundary condition doesn't work internal flow.
2) The order of the P, U.
We'll fix it and let you know

Best regards

Donggyu Kim

**Comment by torodecamargue** [comment]
Hello Donggyu,

How are you? 

Do you know the fan feature will be fixed in the next release of Baram? 
When is the next release? 

Best regards

Julien

**Comment by kdg5424** [comment]
Hello Julien. This is Donggyu. I'm really sorry for the late reply.
The fan boundary condition is ok now. You can use it normally.

Best regards

Donggyu Kim

**Comment by torodecamargue** [comment]
Hello Donggyu,

Thank you for your answer, good news.

Do I need to reinstall an update of the latest version or it was already included in the 24.4 version in september ? 

Best regards

Julien

**Comment by kdg5424** [comment]
Hello Julien.
BARAM is not updated after 24.4 version in September.
We'll upload 25.1 in next year.
And as I said, we updated our BARAM page and uploaded some BARAM tutorial videos on our youtube.
You can see it on our page and youtube

Best regards

Donggyu Kim

**Comment by torodecamargue** [comment]
So it means that the version 24.4 is OK for the fan BC? 

Thank you

**Comment by torodecamargue** [comment]
Hello Donggyu,

Do you know which formulation of the Fan BC uses Baram ? P-Q curve or Q - P curve or U - P curve as you said before in the discussion ? what is U in this case (velocity magnitude) ?

The data of the fan suppliers are pressure in function of volumic flow. 

Do you have a simple example of the use of this BC ? 

Thank you 

BR

Julien

**Reply by kdg5424** [reply]
Hello Julien.

We use U-P Curve. And the unit of the velocity is **m/s**. (Pressure is **Pa**)
And I'm sorry we don't have the tutorial or example for this BC.

**Comment by torodecamargue** [comment]
One more question,

Should I use this BC on an internal BC or on external BC for example instead of an inlet BC ? My system is a closed loop wind tunnel case.
It should match better with the reality if I can use internal BC without inlet and outlet BCs.

One example will be really helpfull, this topic is not clear on the cfd online forum.

I ran a test case with an internal BC but the flow velocity seems to be not realistic 

BR

**Reply by kdg5424** [reply]
You should use this BC on internal BC.
As I said, we don't have the tutorial or example right now.
We'll upload as soon as possible.

Best regards

Donggyu Kim

**Comment by torodecamargue** [comment]
Hello Donggyu,

it works perfectly!

BR

**Comment by kdg5424** [comment]
Hello Julien.

I'm very glad to hear that.
Thank you very much.

Best regards
